src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -399,6 +399,60 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
       else Iterator.empty.next()
   }
 
+  /** Creates an iterator over all the elements of this iterator that
+    *  satisfy the predicate `p`. The order of the elements
+    *  is preserved.
+    *
+    *  '''Note:''' `withFilter` is the same as `filter` on iterators. It exists so that
+    *  for-expressions with filters work over iterators.
+    *
+    *  @param p the predicate used to test values.
+    *  @return  an iterator which produces those values of this iterator which satisfy the predicate `p`.
+    *  @note    Reuse: $consumesAndProducesIterator
+    */
+  def withFilter(p: A => Boolean): Iterator[A] = filter(p)
+
+  /** Creates an iterator over all the elements of this iterator which
+    *  do not satisfy a predicate p.
+    *
+    *  @param p the predicate used to test values.
+    *  @return  an iterator which produces those values of this iterator which do not satisfy the predicate `p`.
+    *  @note    Reuse: $consumesAndProducesIterator
+    */
+  def filterNot(p: A => Boolean): Iterator[A] = filter(!p(_))
```

Comment:
This is an inefficient way to do filtering.  The fast way is to have a `filterImpl` that takes a boolean argument and defer to that with opposite sense for `filter` and `filterNot` (and make `filter` and `filterNot` final).  Maybe it should be called `filterWhen` if it's part of the public API.  Not sure this is the right time to make this change, but I can't think of a better time.